src/wms/src/test/java/org/geoserver/wms/legendgraphic/FeatureCountLegendGraphicTest.java

```diff
@@ -162,6 +162,110 @@ public void testStatesFull() throws Exception {
+    public void testStatesMissingHeightWidth() throws Exception {
+        runGetLegendGraphics(
+                "wms?service=WMS&version=1.1.1&request=GetLegendGraphic&format=image/png"
+                        + "&layer=" + getLayerId(SF_STATES)
+                        + "&style=Population"
+                        + "&srs=EPSG:4326&bbox=" + "-130,24,-66,50"
+                        + "&legend_options="
+                        + GetLegendGraphicRequest.COUNT_MATCHED_KEY + ":true");
+        assertEquals(1, ruleSets.size());
+        Rule[] rules = ruleSets.get(0);
+        logLabels(rules);
+        assertEquals(4, rules.length);
+        assertLabel("2M - 4M (10)", rules[0]);
+        assertLabel("< 2M (16)", rules[1]);
+        assertLabel("> 4M (23)", rules[2]);
+        // this is the rule for outline and text symbolizer, Alaska and Hawaii are not in the
+        // map but Washington DC is and it's not a state (50 - 2 + 1)
+        assertLabel("(49)", rules[3]);
+    }
+
+    @Test
+    public void testStatesMissingHeightWidthSrs() throws Exception {
+        runGetLegendGraphics(
+                "wms?service=WMS&version=1.1.1&request=GetLegendGraphic&format=image/png"
+                        + "&layer=" + getLayerId(SF_STATES)
+                        + "&style=Population"
+                        + "&bbox=" + "-130,24,-66,50"
+                        + "&legend_options="
+                        + GetLegendGraphicRequest.COUNT_MATCHED_KEY + ":true");
+        assertEquals(1, ruleSets.size());
+        Rule[] rules = ruleSets.get(0);
+        logLabels(rules);
+        assertEquals(4, rules.length);
+        assertLabel("2M - 4M (10)", rules[0]);
+        assertLabel("< 2M (16)", rules[1]);
+        assertLabel("> 4M (23)", rules[2]);
+        // this is the rule for outline and text symbolizer, Alaska and Hawaii are not in the
+        // map but Washington DC is and it's not a state (50 - 2 + 1)
+        assertLabel("(49)", rules[3]);
+    }
+
+    @Test
+    public void testStatesMissingBboxSrs() throws Exception {
+        runGetLegendGraphics(
+                "wms?service=WMS&version=1.1.1&request=GetLegendGraphic&format=image/png"
+                        + "&layer=" + getLayerId(SF_STATES)
+                        + "&style=Population"
+                        + "&width=550&height=250"
+                        + "&legend_options="
+                        + GetLegendGraphicRequest.COUNT_MATCHED_KEY + ":true");
+        assertEquals(1, ruleSets.size());
+        Rule[] rules = ruleSets.get(0);
+        logLabels(rules);
+        assertEquals(4, rules.length);
+        assertLabel("2M - 4M (10)", rules[0]);
+        assertLabel("< 2M (16)", rules[1]);
+        assertLabel("> 4M (23)", rules[2]);
+        // this is the rule for outline and text symbolizer, Alaska and Hawaii are not in the
+        // map but Washington DC is and it's not a state (50 - 2 + 1)
+        assertLabel("(49)", rules[3]);
+    }
+
+    /*@Test
```

Comment:
testStatesMissingHeightWidthBboxSrs() fails when activated returning different values for the featureCounts than expected - it does not crash.

The curious thing is that if I run the request used in the test manually (on a jetty:run geoserver web-app instance with GESERVER_DATA_DIR from geoserver/data/release) the request works as expected and returns the values set in asserts in the Test (which are the same than all the test expect). Tested manually with
http://localhost:8080/geoserver/wms?REQUEST=GetLegendGraphic&VERSION=1.1.1&FORMAT=image/png&LAYER=topp:states&style=&legend_options=countMatched:true

The only difference is that I do not have the style param set on manual test, but that can not be the thing. Any idea what may cause this curious behaviour?